tg_bot/modules/sql/locks_sql.py

```diff
@@ -216,6 +231,64 @@ def get_restr(chat_id):
         SESSION.close()
 
 
+def get_whitelist(chat_id):
+    return CHAT_WHITELIST.get(str(chat_id), {})
+
+
+def add_whitelist(chat_id, url):
+    global CHAT_WHITELIST
+    with WHITELIST_LOCK:
+        url = re.search(r'(^http:\/\/|^https:\/\/|^ftp:\/\/|^)(www\.)?(\S*)', url, flags=re.I).group(3).lower()
```

Comment:
given this regex pattern doesnt change, compile it and use it as a global.
Also, what happens if group(3) is None? lower() will die

Reply:
I'm also not convinced this should be here, given it isnt  sql logic (and youre wasting CPU cycles given this bit doesnt need the lock yet)